src/2025h2/parallel-front-end.md

```diff
@@ -0,0 +1,75 @@
+# Promoting Parallel Front End
+
+| Metadata         |                                                   |
+| :--------------- | ------------------------------------------------- |
+| Point of contact | @SparrowLii                                       |
+| Teams            | <!-- TEAMS WITH ASKS -->                          |
+| Task owners      | <!-- TASK OWNERS -->                              |
+| Status           | Proposed                                          |
+| Tracking issue   | [https://github.com/rust-lang/rust/issues/113349] |
+| Zulip channel    | [#t-compiler/wg-parallel-rustc][channel]          |
+
+[channel]: https://rust-lang.zulipchat.com/#narrow/channel/187679-t-compiler.2Fwg-parallel-rustc/
+
+
+## Summary
+
+Continue to parallelize front-end stabilization and performance improvements, continuing from the [2025h1 goal](https://rust-lang.github.io/rust-project-goals/2025h1/parallel-front-end.html).
+
+## Motivation
+
+Verify and resolve the few remaining deadlock and other issues, stabilize the features, and try to further improve the parallel compilation performance through several means.
+
+### The status quo
+
+Parallel front end progressed well in the first half of the year. We resolved the deadlock issue caused by the deadlock handler, added the rustc-rayon  that the parallel front end depends on to the rustc working tree, and optimized the work-stealing scheduling algorithm to avoid deadlocks in the parallelization of rustc queries.
+
+Next we need to resolve and verify the remaining issue list, improve the parallel front end test suit, and include all known issues in the ui tests to ensure the stability of the features.
+
+Then we will start the feature stabilization process to push parallel front end into the stable version.
+
+In addition, there are already contributors working on improving the performance of the parallel front end, such as trying to parallelize the macro expansion process, analysing data contention in query calls, etc. (thank you very much). We will conduct more detailed analysis and implementation of these directions in the second half of the year.
+
+In addition, we need to document and enhance the support of surrounding tools for parallel frontends, such as bootstrap, rustc-perf, and Cargo.
+
+### The next 6 months
+
+- Resolve the remaining issues in the issue [list](https://github.com/rust-lang/rust/issues?q=state%3Aopen%20label%3AA-parallel-compiler). Most of them have been solved, but are lack of ways to ensure stability.
+- Land the parallel front end test [suite](https://github.com/rust-lang/rust/pull/132051) to ensure the robustness and prevent various issues from occurring again.
+- Enable parallel front end in bootstrap.
+- Continue to improve parallel compilation performance, via parallelize the macro expansion and reduce data competition.
+- Enable parallel front end in Cargo.
+- Add more test cases for parallel front end in rustc-perf
```

Comment:
Can add a couple more items
- Enable the parallel frontend by default on nightly, but not on stable, for preliminary testing (similarly to [LLD](https://github.com/rust-lang/rust/pull/140525)) 
- Write stabilization report for the feature and submit it for compiler FCP

Reply:
Thanks! I added the second one. 

And enable it by default (even in nightly) might require more statistics, as it may cause performance regressions compared to single-threaded in many cases.